cmd/skopeo/utils_test.go

```diff
@@ -110,6 +110,29 @@ func TestImageOptionsNewSystemContext(t *testing.T) {
 	assert.Error(t, err)
 }
 
+func TestGlobalOptionsNewSystemContext(t *testing.T) {
+	// Default state
+	opts, _ := fakeGlobalOptions(t, []string{})
+	res := opts.newSystemContext()
+	assert.Equal(t, &types.SystemContext{}, res)
+	// Set everything to non-default values.
+	opts, _ = fakeGlobalOptions(t, []string{
+		"--registries.d", "/srv/registries.d",
+		"--override-arch", "overridden-arch",
+		"--override-os", "overridden-os",
+		"--override-variant", "overridden-variant",
+		"--tmpdir", "/srv",
+	})
+	res = opts.newSystemContext()
+	assert.Equal(t, &types.SystemContext{
+		RegistriesDirPath:    "/srv/registries.d",
+		ArchitectureChoice:   "overridden-arch",
+		OSChoice:             "overridden-os",
+		VariantChoice:        "overridden-variant",
+		BigFilesTemporaryDir: "/srv",
```

Comment:
Tests for `SystemRegistriesConfPath` and `DockerInsecureSkipTLSVerify` are missing.